src/realm/alloc_slab.cpp

```diff
@@ -718,6 +718,8 @@ ref_type SlabAlloc::attach_file(const std::string& file_path, Config& cfg)
             m_baseline = m_file_mappings->m_initial_mapping.get_size();
         }
         ref_type top_ref = 0;
+        // top_ref is useless unless in shared mode as the allocator is not updated to reflect
```

Comment:
What's the relationship between "shared mode" and `cfg.read_only`, if any? The comment makes it sound like that this code should be checking that we're in shared mode before retrieving the top ref, but the code doesn't appear to be doing that.